src/mysql_router_helpers.py

```diff
@@ -157,35 +116,34 @@ def bootstrap_and_start_mysql_router(
             bootstrap_mysqlrouter_command.append("--force")
 
         try:
-            subprocess.run(bootstrap_mysqlrouter_command)
+            subprocess.run(bootstrap_mysqlrouter_command, check=True)
 
-            subprocess.run(f"sudo chmod 755 {MYSQL_HOME_DIRECTORY}/{name}".split())
+            replace_socket_location_commands = [
+                "sudo",
+                "sed",
+                "-Ei",
+                f"s:/tmp/(.+).sock:{CHARMED_MYSQL_COMMON_DIRECTORY}/var/run/mysqlrouter/\\1.sock:g",
+                f"{CHARMED_MYSQL_DATA_DIRECTORY}/etc/mysqlrouter/mysqlrouter.conf",
+            ]
+            subprocess.run(replace_socket_location_commands, check=True)
```

Comment:
`mysqlrouter --bootstrap` does not have a command line flag for socket locations. it defaults to `/tmp`.

in strictly confined snaps, the `/tmp` directory gets mapped to `/tmp/snap-private-tmp/snap.<snap_name>/tmp/` where all parent directories under `/tmp` has permissions `700`. furthermore, the snap cannot define a layout for a top level directory `/tmp`. this results in the socket files being inaccessible by other users

options include:
1. change permissions of directories for the `/tmp` directory -> not good because snap is strictly confined and it feels dirty to change snap generated directories
2. change the location for the socket files after the bootstrap -> done in this PR